hicommonwealth/commonwealth
# Handle hex for new cosmos-evm chains

## Link to Issue
Closes: #6584 
Closes: #6961 
Closes: #7325

## Description of Changes
- New EVM-compatible Cosmos chains will be recognized so keplr-ethereum or Metamask is used
- Refactored the Cosmos Proxy code for clarity and to eliminate unhandled promises.

## "How We Fixed It"

- Adds slip44 column to ChainNodes
  - 60 = evm-compatible node, 118 = standard cosmos coin type
  - queries cosmos.directory for slip44 if not in our DB (handle new chains automatically)
  - Use this differentiator to choose Keplr-Ethereum vs Keplr wallet api
- Reduces sensitivity of Node Liveness system - only flags failed nodes if querying, not making transactions

## Test Plan

- CA (click around) tested on local and:
- run migration
For 6961:
  - at /evmos-dev, [sign in and create a proposal](https://github.com/hicommonwealth/commonwealth/blob/master/knowledge_base/Devnet.md#manually-testing-evmos-dev-ethermint-sandbox) -> expect Keplr "Sign Ethereum" TXs to succeed
  - go to evmos-dev/proposals -> expect proposal to load, no console errors
-  negative test:
  - repeat above steps with a different account
  - expect proposal TX to fail, but previously created proposal still loads (node is not marked as "failed" due to failed tx)


For 6584:
- `update "ChainNodes" set slip44 = null where cosmos_chain_id = 'planq';` in local DB
- go to /planq/proposals
- (may need to refresh page here for next step to work)
- sign in 
  - expect metamask and keplr as wallet options
  - expect Keplr "Sign Ethereum" to work


## Deployment Plan

1. 

## Other Considerations

- Discovered a separate bug, ticketed: https://github.com/hicommonwealth/commonwealth/issues/7084
  - This bug is potential fixed as a side-effect of this PR.
